### PR TITLE
updated doxygen docs for chgres_cube.fd/input_data.F90

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #191 

In preparation for testing, here's an update to the documentation for the code file chgres_cube.fd/input_data.F90.

@GeorgeGayno-NOAA I'm going to need you to review this and add the missing information, particularly the meaning of all the parameters.

Is some of this code no longer used? If so, let's remove it, to save documentation, test, and on-gong maintenance costs.

### CODE REVIEW

**Comment by edwardhartnett** [comment]
@GeorgeGayno-NOAA what you need to do is add my fork as a remote like ths:
git remote add eds_repo git@github.com:edwardhartnett/UFS_UTILS.git
git fetch eds_repo
git checkout ejh_doc_input_data

Then you can edit the file directly to put in the missing documentation.

**Comment by edwardhartnett** [comment]
@GeorgeGayno-NOAA please make this PR a priority so we can continue with UFS_UTILS unit testing tasks.

File: sorc/chgres_cube.fd/input_data.F90 [hunk 1]
```diff
@@ -49,56 +49,56 @@ module input_data
...
+ type(esmf_field)                      :: dpres_input_grid      !< pressure thickness
+ type(esmf_field), public              :: pres_input_grid       !< 3-d pressure
+ type(esmf_field), public              :: ps_input_grid         !< surface pressure
+ type(esmf_field), public              :: terrain_input_grid    !< terrain height
+ type(esmf_field), public              :: temp_input_grid       !< temperature
+ type(esmf_field)                      :: u_input_grid          !< u/v wind at grid
+ type(esmf_field)                      :: v_input_grid          !< box center
+ type(esmf_field), public              :: wind_input_grid       !< 3-component wind
+ type(esmf_field), allocatable, public :: tracers_input_grid(:) !< tracers
+
```
Comment (GeorgeGayno-NOAA) [inline sorc/chgres_cube.fd/input_data.F90:62]: The pound sign gives a very large font in doxygen.  Can you try replacing "#" with "number"

File: sorc/chgres_cube.fd/input_data.F90 [hunk 2]
```diff
@@ -49,56 +49,56 @@ module input_data
...
+ type(esmf_field), public        :: t2m_input_grid          !< 2-m temperature
+ type(esmf_field), public        :: tprcp_input_grid        !< precip
+ type(esmf_field), public        :: ustar_input_grid        !< fric velocity
+ type(esmf_field), public        :: veg_type_input_grid     !< vegetation type
+ type(esmf_field), public        :: z0_input_grid           !< roughness length
+ type(esmf_field), public        :: veg_greenness_input_grid !< vegetation fraction
+ type(esmf_field), public        :: lai_input_grid          !< leaf area index
+ type(esmf_field), public        :: max_veg_greenness_input_grid !< shdmax
+ type(esmf_field), public        :: min_veg_greenness_input_grid !< shdmin
+
```
Comment (GeorgeGayno-NOAA) [inline sorc/chgres_cube.fd/input_data.F90:99]: Remove pound signs?

File: sorc/chgres_cube.fd/input_data.F90 [hunk 3]
```diff
@@ -6355,6 +6371,15 @@ subroutine handle_grib_error(vname,lev,method,value,varnum, iret,var,var8,var3d)
 
 end subroutine handle_grib_error
 
+!> Read GRIB ...
```
Comment (GeorgeGayno-NOAA) [inline sorc/chgres_cube.fd/input_data.F90:6374]: If you want to be more descriptive, you can say "Read in soil temperature and soil moisture from a GRIB2 file."

Comment (edwardhartnett) [reply sorc/chgres_cube.fd/input_data.F90:6374]: Please checkout the branch and update all the doxygen blocks to include at least a sentence or two of description, and document every parameter.

File: sorc/chgres_cube.fd/input_data.F90 (hunk 2 above)
Comment (edwardhartnett) [reply sorc/chgres_cube.fd/input_data.F90:99]: Can you do this on the branch?

File: sorc/chgres_cube.fd/input_data.F90 [hunk 4]
```diff
@@ -717,11 +720,10 @@ subroutine init_atm_esmf_fields
...
  end subroutine init_atm_esmf_fields
 
-!---------------------------------------------------------------------------
-! Read input atmospheric data from spectral gfs (old sigio format).
-! Used prior to July 19, 2017.
-!---------------------------------------------------------------------------
-
+!> Read input atmospheric data from spectral gfs (old sigio format).
+!! Used prior to July 19, 2017.
+!! @param localpet
```
Comment (edwardhartnett) [inline sorc/chgres_cube.fd/input_data.F90:725]: WHat is param localpet? It appears in many places.

File: sorc/chgres_cube.fd/input_data.F90 [hunk 5]
```diff
@@ -5958,6 +5946,15 @@ subroutine read_input_nst_nemsio_file(localpet)
 
  end subroutine read_input_nst_nemsio_file
 
+!> Read 
+!! @param field
```
Comment (edwardhartnett) [inline sorc/chgres_cube.fd/input_data.F90:5950]: Please put in a one or two sentence description of each param.

File: sorc/chgres_cube.fd/input_data.F90 [hunk 6]
```diff
@@ -5996,10 +5993,13 @@ SUBROUTINE READ_FV3_GRID_DATA_NETCDF(FIELD,TILE_NUM,IMO,JMO,LMO, &
 
  END SUBROUTINE READ_FV3_GRID_DATA_NETCDF
  
- !---------------------------------------------------------------------------
-! Read winds from a grib2 file
-!---------------------------------------------------------------------------
-
+!> Read winds from a grib2 file.
+!! @param file
```
Comment (edwardhartnett) [inline sorc/chgres_cube.fd/input_data.F90:5997]: Similarly, document the params.

**Review by edwardhartnett — COMMENTED** [review]
OK, I've made the necessary changes and this PR is ready to merge.

**Comment by GeorgeGayno-NOAA** [comment]
Here is what I see as the final result: https://georgegayno-noaa.github.io/UFS_UTILS/classinput__data.html

Some of the routines in input_data.F90 are not listed.  For example, routine 'read_grib_soil'.  I did not see any errors or warnings when the documentation was built.  Not sure what is going on.

**Comment by edwardhartnett** [comment]
Stand by, I will check...

**Comment by edwardhartnett** [comment]
Also after that last merge and update, the doxygen warnings will be available in the GitHub action build...

**Comment by GeorgeGayno-NOAA** [comment]
> Also after that last merge and update, the doxygen warnings will be available in the GitHub action build...

Ok.  I will try again.

**Comment by GeorgeGayno-NOAA** [comment]
I see the doxygen warnings.  Other than a few undefined variables, there is nothing that indicates why certain routines in input_data.F90 are not listed.  If you look at the source file (https://georgegayno-noaa.github.io/UFS_UTILS/input__data_8F90_source.html), the doxygen comments for "read_grib_soil" are not there.  So, I assume doxygen recognized them.   Let me try something else.

**Comment by edwardhartnett** [comment]
When I build these changes locally, all functions are included, including read_grib_soil().

Let's merge this and circle around to input_data again after getting some of the other outstanding documentation PRs done.

**Comment by GeorgeGayno-NOAA** [comment]
> When I build these changes locally, all functions are included, including read_grib_soil().
> 
> Let's merge this and circle around to input_data again after getting some of the other outstanding documentation PRs done.

I was building on Dell.  Sometimes the builds are a bit different on each machine (like Dell vs. Hera).

**Review by GeorgeGayno-NOAA — APPROVED** [review]

**Pull request merged**